### ISSUE TITLE
[#1945 Follow-up] 

### ISSUE THREAD

**Issue by ctlkkc** [olifolkerd/tabulator#2246]
Hey Oli,

Thanks for adding the great feature of htmlOutputConfig:columnCalcs, but I seem not able to get the bottomCalcs row after setting the table as documented. Here is my codes:

```
	<link href="./libs/tabulator.4.3.0.css" rel="stylesheet">
	<script src="./libs/tabulator.min.4.3.0.js"></script>
......
				table2 = new Tabulator("#table2", {
					height: "100%",
					layout: "fitColumns",
					clipboard: "copy",
					clipboardCopyHeader: true,
					clipboardCopySelector: "active",
					htmlOutputConfig:{
							columnGroups:true, //do not include column groups in column headers for HTML table
							rowGroups:true, //do not include row groups in HTML table
							columnCalcs:true, //do not include column calcs in HTML table
					},
					ajaxResponse: function (url, params, response) {
						$("#table2_timestamp_text").val('LAST UPDATED: ' + moment(response[0]).clone().tz(moment.tz.guess()).format("YYYY-MM-DD hh:mm:ss A"));
						response.shift();
						document.getElementById('status').setAttribute('style', 'background-color: #00CD00;');
						$("#status_text").val(' UPDATING');
						return response;
					},
					ajaxError: function (xhr, textStatus, errorThrown) {
						document.getElementById('status').setAttribute('style', 'background-color: #FF1919;');
						$("#status_text").val(' DISCONNECTED');
					},
					columns: [
						{ title: "CLIENT_NAME", field: "ClientName", headerFilter: "input" },
						{ title: "CLIENT_ID", field: "ClientId", headerFilter: "input", bottomCalc: Total },
						{ title: "ATHENA_EXEC_CNT", field: "Athena Exec Cnt", sorter: "number",   headerFilter: "input", headerFilterPlaceholder: ">=", headerFilterFunc: customHeaderFilter, cellClick: function (e, cell) { CellAction('Asia/Tokyo', "stats_vs_flux", cell.getRow().getData().ClientId, "Athena Exec Cnt") }, bottomCalc: Sum },
						{ title: "ATHENA_TOTLEXECQTY", field: "Athena TotlExecQty", sorter: "number",   headerFilter: "input", headerFilterPlaceholder: ">=", headerFilterFunc: customHeaderFilter, cellClick: function (e, cell) { CellAction('Asia/Tokyo', "stats_vs_flux", cell.getRow().getData().ClientId, "Athena TotlExecQty") }, bottomCalc: Sum },
						{ title: "ATHENA_EXECTRANSCNT", field: "Athena ExecTransCnt", sorter: "number",   headerFilter: "input", headerFilterPlaceholder: ">=", headerFilterFunc: customHeaderFilter, cellClick: function (e, cell) { CellAction('Asia/Tokyo', "stats_vs_flux", cell.getRow().getData().ClientId, "Athena ExecTransCnt") }, bottomCalc: Sum },
						{ title: "FLUX_EXEC_CNT", field: "Flux Exec Cnt", sorter: "number",   headerFilter: "input", headerFilterPlaceholder: ">=", headerFilterFunc: customHeaderFilter, cellClick: function (e, cell) { CellAction('Asia/Tokyo', "stats_vs_flux", cell.getRow().getData().ClientId, "Flux Exec Cnt") }, bottomCalc: Sum },
						{ title: "FLUX_TOTLEXECQTY", field: "Flux TotlExecQty", sorter: "number",   headerFilter: "input", headerFilterPlaceholder: ">=", headerFilterFunc: customHeaderFilter, cellClick: function (e, cell) { CellAction('Asia/Tokyo', "stats_vs_flux", cell.getRow().getData().ClientId, "Flux TotlExecQty") }, bottomCalc: Sum },
						{ title: "FLUX_ORDER_CNT", field: "Flux Ord Cnt", sorter: "number",   headerFilter: "input", headerFilterPlaceholder: ">=", headerFilterFunc: customHeaderFilter, cellClick: function (e, cell) { CellAction('Asia/Tokyo', "stats_vs_flux", cell.getRow().getData().ClientId, "Flux Ord Cnt") }, bottomCalc: Sum },
						{ title: "FLUX_ORDTRANSCNT", field: "Flux OrdTransCnt", sorter: "number",   headerFilter: "input", headerFilterPlaceholder: ">=", headerFilterFunc: customHeaderFilter, cellClick: function (e, cell) { CellAction('Asia/Tokyo', "stats_vs_flux", cell.getRow().getData().ClientId, "Flux OrdTransCnt") }, bottomCalc: Sum },
						{ title: "FLUX_GROSS_VALUE", field: "Flux Gross Value", sorter: "number",   headerFilter: "input", headerFilterPlaceholder: ">=", headerFilterFunc: customHeaderFilter, cellClick: function (e, cell) { CellAction('Asia/Tokyo', "stats_vs_flux", cell.getRow().getData().ClientId, "Flux Gross Value") }, bottomCalc: Sum },
						{ title: "FLUX_GROSS_EXECUTED_VALUE", field: "Flux Gross Executed Value", sorter: "number",   headerFilter: "input", headerFilterPlaceholder: ">=", headerFilterFunc: customHeaderFilter, cellClick: function (e, cell) { CellAction('Asia/Tokyo', "stats_vs_flux", cell.getRow().getData().ClientId, "Flux Gross Executed Value") }, bottomCalc: Sum },
						{
							title: "DIFF_IN_EXEC_CNT", field: "Diff In Exec Count", sorter: "number", headerFilter: "input", headerFilterPlaceholder: ">=", headerFilterFunc: customHeaderFilter, formatter: function (cell, formatterParams) {
								var cellValue = cell.getValue();
								if (cellValue === 'Match' || cellValue === 'N/A') {
									cell.getElement().style.backgroundColor = "#98f677";
									return cellValue;
								} else if (Number(cellValue) > 0) {
									cell.getElement().style.backgroundColor = "#ff4c4c";
									return cellValue;
								} else {
									return cellValue.toLocaleString();
								}
							}, bottomCalc: Diff
						},
						{
							title: "DIFF_IN_EXEC_QTY", field: "Diff In Exec Qty", sorter: "number", headerFilter: "input", headerFilterPlaceholder: ">=", headerFilterFunc: customHeaderFilter, formatter: function (cell, formatterParams) {
								var cellValue = cell.getValue();
								if (cellValue === 'Match' || cellValue === 'N/A') {
									cell.getElement().style.backgroundColor = "#98f677";
									return cellValue;
								} else if (Number(cellValue) > 0) {
									cell.getElement().style.backgroundColor = "#ff4c4c";
									return cellValue;
								} else {
									return cellValue.toLocaleString();
								}
							}, bottomCalc: Diff
						}
					]
				});
				table2.setData("/s_c.pass.stats_athena_vs_flux", {}, ajaxConfig);
......
var table_message = eval(table).getHtml(); //table="table2"
console.log(table_message);
```

And here is what I've got from the console:
```
<table class="tabulator-print-table"><thead><tr><th colspan="1" rowspan="1">CLIENT_NAME</th><th colspan="1" rowspan="1">CLIENT_ID</th><th colspan="1" rowspan="1">ATHENA_EXEC_CNT</th><th colspan="1" rowspan="1">ATHENA_TOTLEXECQTY</th><th colspan="1" rowspan="1">ATHENA_EXECTRANSCNT</th><th colspan="1" rowspan="1">FLUX_EXEC_CNT</th><th colspan="1" rowspan="1">FLUX_TOTLEXECQTY</th><th colspan="1" rowspan="1">FLUX_ORDER_CNT</th><th colspan="1" rowspan="1">FLUX_ORDTRANSCNT</th><th colspan="1" rowspan="1">FLUX_GROSS_VALUE</th><th colspan="1" rowspan="1">FLUX_GROSS_EXECUTED_VALUE</th><th colspan="1" rowspan="1">DIFF_IN_EXEC_CNT</th><th colspan="1" rowspan="1">DIFF_IN_EXEC_QTY</th></tr></thead><tbody><tr class="tabulator-print-table-row"><td>CLIENT</td><td>11,303,983</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td style="background-color: rgb(152, 246, 119);">Match</td><td style="background-color: rgb(152, 246, 119);">Match</td></tr><tr class="tabulator-print-table-row"><td>CLIENT</td><td>11,328,662</td><td>17,845</td><td>1,704,779</td><td>35,690</td><td>17,855</td><td>N/A</td><td>1,202,488</td><td>3,919,843</td><td>3,337,628,710</td><td>N/A</td><td>-10</td><td style="background-color: rgb(152, 246, 119);">N/A</td></tr><tr class="tabulator-print-table-row"><td>CLIENT</td><td>11,333,292</td><td>2,163</td><td>564,505</td><td>2,163</td><td>2,164</td><td>564,605</td><td>15,994</td><td>63,053</td><td>661,943,700</td><td>650,726,400</td><td>-1</td><td>-100</td></tr><tr class="tabulator-print-table-row"><td>CLIENT</td><td>11,338,239</td><td>3,895</td><td>453,429</td><td>7,544</td><td>3,895</td><td>453,429</td><td>622,143</td><td>2,534,109</td><td>2,398,767,736</td><td>422,493,066</td><td style="background-color: rgb(152, 246, 119);">Match</td><td style="background-color: rgb(152, 246, 119);">Match</td></tr><tr class="tabulator-print-table-row"><td>CLIENT</td><td>11,342,911</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td style="background-color: rgb(152, 246, 119);">Match</td><td style="background-color: rgb(152, 246, 119);">Match</td></tr><tr class="tabulator-print-table-row"><td>CLIENT</td><td>11,343,758</td><td>1,544</td><td>613,057</td><td>1,544</td><td>1,544</td><td>613,057</td><td>1,545</td><td>4,636</td><td>1,126,373,970</td><td>1,125,574,500</td><td style="background-color: rgb(152, 246, 119);">Match</td><td style="background-color: rgb(152, 246, 119);">Match</td></tr><tr class="tabulator-print-table-row"><td>CLIENT</td><td>11,343,803</td><td>796</td><td>456,839</td><td>796</td><td>796</td><td>456,839</td><td>1,435</td><td>4,327</td><td>268,201,622</td><td>268,201,622</td><td style="background-color: rgb(152, 246, 119);">Match</td><td style="background-color: rgb(152, 246, 119);">Match</td></tr><tr class="tabulator-print-table-row"><td>CLIENT</td><td>11,344,032</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td style="background-color: rgb(152, 246, 119);">Match</td><td style="background-color: rgb(152, 246, 119);">Match</td></tr><tr class="tabulator-print-table-row"><td>CLIENT</td><td>11,344,399</td><td>28,635</td><td>8,163,978</td><td>32,222</td><td>28,644</td><td>8,165,778</td><td>490,780</td><td>2,313,885</td><td>24,221,862,830</td><td>14,870,535,720</td><td>-9</td><td>-1,800</td></tr><tr class="tabulator-print-table-row"><td>CLIENT</td><td>11,345,194</td><td>2,263</td><td>354,708</td><td>2,263</td><td>2,263</td><td>354,708</td><td>3,334</td><td>12,231</td><td>640,615,280</td><td>621,070,780</td><td style="background-color: rgb(152, 246, 119);">Match</td><td style="background-color: rgb(152, 246, 119);">Match</td></tr></tbody></table>
```

It lacks the row of bottomCalcs as displayed.
Can you help telling what cause this? 

Best,

**Comment by aditya-thimphu** [comment]
Hi,

I'm having a similar issue with regard to printing the table. Print page does not include table footer column calc. Also alignment of numbers which is set to right does not appear right aligned in the print.

Please check [JS fiddle here](https://jsfiddle.net/wmbhtzag/1/). I have also discussed the same issue in StackOverflow [here](https://stackoverflow.com/questions/57158498/tabulators-output-of-print-function-does-not-include-table-footer-column-calcul).

Thanks

**Comment by olifolkerd** [comment]
Hey,

It turns out i had covered the group level calculation rows but not the table level calculation rows.

I have pushed a fix for this to the 4.4 branch and will include it in todays release.

Cheers

Oli :)
